code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -395,25 +395,21 @@
 			if(valve_open)
 				logmsg = "Valve was <b>opened</b> by [key_name(usr)], starting a transfer into \the [holding || "air"].<br>"
 				if(!holding)
-					var/plasma = air_contents.gases["plasma"]
-					var/n2o = air_contents.gases["n2o"]
-					var/bz = air_contents.gases["bz"]
-					var/freon = air_contents.gases["freon"]
-					if(n2o || plasma || bz || freon)
+					var/list/danger = list()
+					for(var/id in air_contents.gases)
+						var/gas = air_contents.gases[id]
+						if(!gas[GAS_META][META_GAS_DANGER])
+							continue
+						if(gas[MOLES] > (gas[GAS_META][META_GAS_MOLES_VISIBLE] || MOLES_PLASMA_VISIBLE)) //if moles_visible is undefined, default to plasma visibility
+							danger[gas[GAS_META][META_GAS_NAME]] = gas[MOLES] //ex. "plasma" = 20
+
+					if(danger.len)
 						message_admins("[ADMIN_LOOKUPFLW(usr)] opened a canister that contains the following: [ADMIN_JMP(src)]")
 						log_admin("[key_name(usr)] opened a canister that contains the following at [COORD(src)]:")
-						if(plasma)
-							log_admin("Plasma")
-							message_admins("Plasma")
-						if(n2o)
-							log_admin("N2O")
-							message_admins("N2O")
-						if(bz)
-							log_admin("BZ Gas")
-							message_admins("BZ Gas")
-						if(freon)
-							log_admin("Freon")
-							message_admins("Freon")
+						for(var/name in danger)
```

Comment:
Please can you just use `english_list` rather than having multiline admin message